python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -128,6 +135,25 @@ def notify_resource_usage(
     ) -> None:
         free_slots = self._actor_pool.num_free_slots()
         if input_queue_size > free_slots and under_resource_limits:
+            # The first time we request actors, we record the timestamp.
+            if self._actors_requested_ts is None:
+                self._actors_requested_ts = time.time()
+
+            # If the minimum requested actors are not available in a timely manner,
+            # raise a timeout.
+            exceeds_timeout = (
+                time.time() - self._actors_requested_ts
+                > DEFAULT_WAIT_FOR_MIN_ACTORS_SEC
+            )
+            if (
```

Comment:
we should also check if `self._actor_pool.num_running_actors() >= self._autoscaling_policy.min_workers`, if yes, then we don't need to check any more.

Reply:
I don't think we should wait here, only on the first start. There should be a place that prints out waiting for initial actors to start.

We wouldn't want to timeout jobs in the middle, only at the beginning, to be conservative.

The change should be adding a single timeout to ray.get I believe.

Reply:
@ericl - yeah make sense to me.